src/mesh/index_derivs.cxx

```diff
@@ -2767,7 +2767,7 @@ const Field3D Mesh::indexFDDX(const Field3D &v, const Field3D &f, CELL_LOC outlo
 
   if (this->xstart > 1) {
     // Two or more guard cells
-    if (StaggerGrids) {
+    if (StaggerGrids && vloc != diffloc) {
```

Comment:
This line seems to be equivalent to that on line 2736 above -- I think this means we should never be able to hit the throw just below (as we'd already have thrown). Think the main practical application of this is that the nested `if` below only needs to care about one of `vloc` and `diffloc` (although having both may help make things clear).

Reply:
`diffloc` is changed to `inloc` at those lines though? When is `diffloc` ever not `inloc`?

What combinations of locations were causing problems?

Reply:
> It's not equivalent: at line 2736 diffloc=inloc, but diffloc can be changed at 2742 or 2747. I actually added this because in testing combinations of vloc/inloc/outloc that should throw, some did not without this condition.

But I think the possible changes at 2742 and 2747 force `diffloc = inloc` according to the logic in the `if`s